tulip/abstract/discretization.py

```diff
@@ -983,6 +1070,474 @@ def discretize(
+        orig_list = []
+        for poly in part:
+            if len(poly) == 0:
+                orig_list.append(poly.copy())
+            elif len(poly) == 1:
+                orig_list.append(poly[0].copy())
+            else:
+                raise Exception("discretize: "
+                    "problem in convexification")
+        orig = list(range(len(orig_list)))
+
+    # Cheby radius of disturbance set
+    # (defined within the loop for pwa systems)
+    if islti:
+        if len(ssys.E) > 0:
+            rd = ssys.Wset.chebR
+        else:
+            rd = 0.
+
+    # Initialize matrix for pairs to check
+    IJ = part.adj.copy()
+    IJ = IJ.todense()
+    IJ = np.array(IJ)
+    logger.debug("\n Starting IJ: \n" + str(IJ) )
+
+    # next line omitted in discretize_overlap
+    IJ = reachable_within(trans_length, IJ,
+                          np.array(part.adj.todense()))
+
+    # Initialize output
+    num_regions = len(part)
+    transitions = np.zeros(
+        [num_regions, num_regions],
+        dtype = int
+    )
+    sol = deepcopy(part.regions)
+    adj = part.adj.copy()
+    adj = adj.todense()
+    adj = np.array(adj)
+
+    # next 2 lines omitted in discretize_overlap
+    if ispwa:
+        subsys_list = list(ppp2pwa)
+    else:
+        subsys_list = None
+    ss = ssys
+
+    # init graphics
+    if plotit:
+        try:
+            import matplotlib.pyplot as plt
+
+            plt.ion()
+            fig, (ax1, ax2) = plt.subplots(1, 2)
+            ax1.axis('scaled')
+            ax2.axis('scaled')
+            file_extension = 'pdf'
+        except:
+            logger.error('failed to import matplotlib')
+            plt = None
+    else:
+    	plt = None
+
+    iter_count = 0
+
+    # List of how many "new" regions
+    # have been created for each region
+    # and a list of original number of neighbors
+    #num_new_reg = np.zeros(len(orig_list))
+    #num_orig_neigh = np.sum(adj, axis=1).flatten() - 1
+
+    progress = list()
+        
+    # Do the abstraction
+    while np.sum(IJ) > 0:
+        ind = np.nonzero(IJ)
+        # i,j swapped in discretize_overlap
+        i = ind[1][0]
+        j = ind[0][0]
+        IJ[j, i] = 0
+        si = sol[i]
+        sj = sol[j]
+
+        si_tmp = deepcopy(si)
+        sj_tmp = deepcopy(sj)
+
+        #num_new_reg[i] += 1
+        #print(num_new_reg)
+
+        if ispwa:
+            ss = ssys.list_subsys[subsys_list[i]]
+            if len(ss.E) > 0:
+                rd, xd = pc.cheby_ball(ss.Wset)
+            else:
+                rd = 0.
+
+        if conservative:
+            # Don't use trans_set
+            trans_set = None
+        else:
+            # Use original cell as trans_set
+            trans_set = orig_list[orig[i]]
+
+        S0 = solve_feasible(
+            si, sj, ss, N, closed_loop,
+            use_all_horizon, trans_set, max_num_poly
+        )
+
+        msg = '\n Working with partition cells: ' + str(i) + ', ' + str(j)
```

Comment:
Please consider using the method `str.format`, because it leads to more readable substitutions. For example:

```python
msg = '\n Working with partition cells: {i}, {j}'.format(
    i=i, j=j)
```

The method `str.format` is especially convenient in language-related operations, for example code generation.